DB/src/main/java/io/deephaven/db/v2/sources/chunkcolumnsource/ChunkColumnSource.java

```diff
@@ -29,6 +45,18 @@
         }
     }
 
+    /**
+     * Append a chunk of data to this column source.
+     *
+     * @param chunk the chunk of data to add
+     */
     void addChunk(Chunk<? extends Attributes.Values> chunk);
+
+    /**
+     *  Reset the column source to be ready for reuse.
+     *
+     *  Cear will discard the currently held chunks.  This should not be  called if a table will continue to reference
+     *  the column source; as it violates the immutability contract.
+     */
     void clear();
```

Comment:
The clear implementations should also return chunks to the pool. See my previous claims about `PoolableChunk`; what should we do about it?

Reply:
WritableChunk is not pretty, but the path of least resistance.